src/Microsoft.DocAsCode.Common/CommandUtility.cs

```diff
@@ -116,21 +117,35 @@ public static int RunCommand(CommandInfo commandInfo, StreamWriter stdoutWriter
         public static bool ExistCommand(string commandName)
         {
             int exitCode;
-            if (Environment.OSVersion.Platform == PlatformID.Unix || Environment.OSVersion.Platform == PlatformID.MacOSX)
+            var errorStream = new MemoryStream();
+            var outputStream = new MemoryStream();
+            using (var outputWriter = new StreamWriter(outputStream))
+            using (var errorWriter = new StreamWriter(errorStream))
             {
-                exitCode = RunCommand(new CommandInfo
+                if (Environment.OSVersion.Platform == PlatformID.Unix || Environment.OSVersion.Platform == PlatformID.MacOSX)
                 {
-                    Name = "type",
-                    Arguments = commandName
-                }, timeoutInMilliseconds: 1000);
-            }
-            else
-            {
-                exitCode = RunCommand(new CommandInfo
+                    exitCode = RunCommand(new CommandInfo
+                    {
+                        // type is a bash command, hence should be an argument to 'bash'
+                        Name = "bash",
+                        Arguments = $"-c \"type {commandName}\""
+                    }, outputWriter, errorWriter, timeoutInMilliseconds: 1000);
+                }
+                else
                 {
-                    Name = "where",
-                    Arguments = commandName
-                }, timeoutInMilliseconds: 1000);
+                    exitCode = RunCommand(new CommandInfo
+                    {
+                        Name = "where",
+                        Arguments = commandName
+                    }, outputWriter, errorWriter, timeoutInMilliseconds: 1000);
+                }
+
+                String resultError = System.Text.Encoding.UTF8.GetString(errorStream.ToArray(), 0, (int)errorStream.Length);
+                String resultOutput = System.Text.Encoding.UTF8.GetString(outputStream.ToArray(), 0, (int)outputStream.Length);
+                if (!string.IsNullOrEmpty(resultError) || !string.IsNullOrEmpty(resultOutput))
+                {
+                    Logger.LogWarning($"RunCommand exitCode: {exitCode}, output: \"{resultOutput}\", error: \"{exitCode}\"");
```

Comment:
>LogWarning [](start = 27, length = 10)

warning looks too heavy for checking command existence

Reply:
Same comment as above.

Reply:
I will have to fix the last `exitCode` reference to  `resultError`.

This warning message would only appear if there's an error / warning when the command is run. 

I think it is important to know what the fault is, if the program is able to tell us so.

Comments?

Reply:
Similar to above, would it be better to rewrite it as
```
Logger.LogDiagnostic($"RunCommand exitCode: {exitCode}, output: \"{resultOutput}\", error: \"{resultError}\"");
```

Reply:
For example, When `git` does not exist, `where git` prints to `resultError`. however, I don't think docfx should print warning, as `git` not exist is also an acceptable scenario. `LogDiagnostic` sounds better.

Reply:
I make a change in https://github.com/dotnet/docfx/pull/2090 so that it does not throw exception every time when git does not exist, after the change, my concern about `warning too much` is gone and it looks cool to me to use `LogWarning`

Reply:
Looking forward to #2090 being merged. I will retain LogWarning for this change.